manager/integration/tests/test_csi_snapshotter.py

```diff
@@ -674,23 +732,213 @@ def test_csi_snapshot_with_snap_param(client, volume_name): # NOQA
+                                         name=deployment_name)
+    vol = common.wait_for_volume_detached(client, vol.name)
+
+    csivolsnap = volumesnapshot(vol.name + "-volumesnapshot",
+                                "default",
+                                csisnapclass["metadata"]["name"],
+                                "persistentVolumeClaimName",
+                                pvc_name)
+
+    wait_for_volumesnapshot_ready(
+                            volumesnapshot_name=csivolsnap["metadata"]["name"],
+                            namespace='default',
+                            ready_to_use=False)
+
+    # Volume is in attached state
+    deployment['spec']['replicas'] = 1
+    apps_api.patch_namespaced_deployment(body=deployment,
+                                         namespace='default',
+                                         name=deployment_name)
+    vol = common.wait_for_volume_attached(client, vol.name)
+
+    wait_for_volumesnapshot_ready(
+                            volumesnapshot_name=csivolsnap["metadata"]["name"],
+                            namespace='default',
+                            ready_to_use=True)
+
+
+def test_csi_snapshot_snap_create_volume_from_snapshot(apps_api, # NOQA
+                                      client, # NOQA
+                                      make_deployment_with_pvc, # NOQA
+                                      volume_name, # NOQA
+                                      volumesnapshotclass, # NOQA
+                                      volumesnapshot, # NOQA
+                                      csi_pv, # NOQA
+                                      pvc, # NOQA
+                                      core_api, # NOQA
+                                      pod_make): # NOQA
+    """
+    Context:
+
+    After deploy the CSI snapshot CRDs, Controller at
+    https://longhorn.io/docs/1.2.4/snapshots-and-backups/
+    csi-snapshot-support/enable-csi-snapshot-support/
+
+    Create VolumeSnapshotClass with type=snap
+      - longhorn-snapshot (type=snap)
+
+    Test the extend CSI snapshot type=snap support to Longhorn snapshot
+
+    Steps:
+
+    0. Create Longhorn volume test-vol
+        - Size 5GB
+        - Create PV/PVC/Workload for the Longhorn volume
+        - Write data into volume
+        - Setup backup store
+
+    1. Test create new volume from CSI snapshot
+        - Create VolumeSnapshot with class longhorn-snap
         - Create volume from longhorn-snapshot
             - Source volume is attached && Longhorn snapshot exist
                 - Create PVC from snapshot generated from step 1
                 - Attach the PVC and verify data
             - Source volume is detached
                 - Scale down the workload
-                - Create PVC from VolumeSnapshot generated from step 1
+                - Create PVC from VolumeSnapshot generated from step beggining
                 - Verify PVC provision failed
```

Comment:
Can we add the steps:
* Scale up the workload
* Wait for PVC to finish provisioning and be bounded
* Attach the PVC test-restore-pvc and verify the data

Reply:
HI @PhanLe1010 
I added steps into code, thank you